core-bundle/src/Security/Authentication/Provider/AuthenticationProvider.php

```diff
@@ -123,9 +123,6 @@ public function authenticate(TokenInterface $token): TokenInterface
 
         // Skip two-factor authentication on trusted devices
         if ($this->trustedDeviceManager->isTrustedDevice($user, $firewallName)) {
```

Comment:
The `$context` is not actually needed if we have trusted devices. we can simply use `$token->getUser()` and `$this->providerKey` for the `isTrustedDevice` call, and return `$token` if successful. We only need the context stuff below to call `beginTwoFactorAuthentication`. Maybe this could be optimized?

Reply:
Like in 47129352d?